packages/store-ui/src/atoms/Price/Price.tsx

```diff
@@ -0,0 +1,49 @@
+import type { HTMLAttributes, ElementType, ReactNode } from 'react'
+import React, { forwardRef } from 'react'
+
+export type PriceFormatter = (price: number, variant: PriceVariant) => ReactNode
+
+export type PriceVariant =
+  | 'default'
+  | 'listing'
+  | 'spot'
+  | 'savings'
+  | 'installment'
+
+export interface PriceProps
+  extends Omit<HTMLAttributes<HTMLSpanElement>, 'children'> {
+  as?: ElementType
+  testId?: string
+  value: number
+  formatter?: PriceFormatter
+  variant?: PriceVariant
+}
+
+export const Price = forwardRef<Omit<HTMLSpanElement, 'children'>, PriceProps>(
+  function Price(
+    {
+      as: Component = 'span',
+      testId = 'store-price',
+      value,
+      formatter = (price) => price,
+      variant = 'default',
+      ...rawProps
+    },
+    ref
+  ) {
+    const formattedPrice = formatter(value, variant)
+
+    const props = {
+      [`data-store-price-${variant}`]: true,
```

Comment:
Can't we use the other convention we were using, like:
```suggestion
      [`data-${variant}`]: true,
```

Reply:
yes, it's a pseudo data attribute!

Reply:
What made me do it this way is that I fear that, in the future, we overlap `data-${variant}` attributes between atoms or organisms that have the same variations, which can lead to styling issues if people are selecting these elements just by the `data-${variant}` selector. Do you think we should worry about this?

Reply:
This shouldn't happen, because this is a pseudo-class, so it's always associated with a class and never alone. Like this:
### Good
```css
[data-store-price][data-listing] {
  // styles
}
```
### Bad
```css
[data-store-price] {
  // styles
}
[data-store-price-listing] {
  // styles
}
```

Reply:
So CSS guarantees users can't do the following?

```css
[data-listing] {
  // styles
}
```

Reply:
yes, he can, but is not a best practice. We do not block the user possibilities but educate them.

Reply:
I agree. But I also think we should build a dummy-proof API if we can. I think `data-store-price-listing` is too long, but it comes with the benefit that it's almost impossible to overlap with other pseudo-class.

I think we would be blocking users more if we only advise them to use the `Good` example. I don't think it's unreasonable to use only `data-error` (in the case of inputs) if the user only wants to set the border color for that one variation. If users can do it, they will do it. By setting it as `data-store-input-error`, we assure that there'll be no overlap between pseudo-classes.

I personally think the `Bad` example is more readable than the `Good` one.

Reply:
Nice, yes. I agree with @emersonlaurentino , we should always style like
```
[data-store-price][data-listing] {
  // styles
}```